src/app/pages/LandingPage.svelte

```diff
@@ -6,21 +6,31 @@
   import DownloadKMP from "../components/DownloadKMP.svelte";
   import SplitButton from "../components/SplitButton.svelte";
   import { currentDownloadURL } from "../stores";
+  import type { KeyboardMetadata } from "@common/types";
+  import type { RelevantKmpOptions } from "@common/kmp-json-file";
 
-  let languageStatus: boolean = false;
+  let selectedLanguage: KeyboardMetadata | undefined = undefined;
   let continueReady: boolean = false;
   let uploadFile: boolean = true;
 
-  $: continueReady = languageStatus && Boolean($currentDownloadURL);
+  $: continueReady =
+    selectedLanguage !== undefined && Boolean($currentDownloadURL);
 
-  function setLanguage(event: CustomEvent) {
-    const languages = event.detail.value;
-    languageStatus = event.detail.status;
-    if (languageStatus) {
-      worker.setProjectData({ languages });
+  $: if (Boolean($currentDownloadURL)) {
+    if (selectedLanguage === undefined) {
+      selectedLanguage = { language: "Undefined Language", bcp47Tag: "und" };
     }
   }
 
+  $: if (selectedLanguage !== undefined) {
+    const options: Partial<Readonly<RelevantKmpOptions>> = {
+      languages: [
+        { name: selectedLanguage.language, id: selectedLanguage.bcp47Tag },
+      ],
+    };
+    worker.setProjectData(options);
+  }
```

Comment:
When a reactive statement gets too large, it's best to Extract Function and pass  that function the reactive variable that might change. For example,

```suggestion
  $: updateLanguage(selectedLanguage)

  function updateLanguage(lang: KeyboardMetadata): void {
    if (lang === undefined) return;
    const options: Partial<Readonly<RelevantKmpOptions>> = {
      languages: [
        { name: lang.language, id: lang.bcp47Tag },
      ],
    };
    worker.setProjectData(options);
  }
```